packages/astro/src/core/render/dev/css.ts

```diff
@@ -30,8 +33,13 @@ export function getStylesForURL(filePath: URL, viteServer: vite.ViteDevServer):
 				continue;
 			}
 			if (id === entry.id) {
+				let upToDateEntry = entry;
+				if (!entry.ssrTransformResult) {
```

Comment:
can you add a comment explaining why you do this, for future readers? Given the "magic incantation" vibes we're starting to get from this function, we should probably comment everything here very liberally.